src/Symfony/Component/Validator/Constraints/PwnedValidator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\RuntimeException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Checks if a password has been leaked in a data breach using haveibeenpwned.com's API.
+ * Use a k-anonymity model to protect the password being searched for.
+ *
+ * @see https://haveibeenpwned.com/API/v2#SearchingPwnedPasswordsByRange
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PwnedValidator extends ConstraintValidator
+{
+    private const RANGE_API = 'https://api.pwnedpasswords.com/range/%s';
+    private $httpClient;
+
+    /**
+     * @param callable|null $httpClient Must mimic the file_get_contents's signature: function(string $url): string|false (false is returned in case of error)
```

Comment:
That's probably the worse API for an HTTP client (in case of error, you won't know what kind of error it is). Btw, mimicking the `file_get_contents` would then mean using `trigger_error` to report details about the failure...

Reply:
@stof I can change that, but it's mostly to be able to mock `file_get_contents` in tests.

Reply:
@stof do you have an API suggestion (something lightweight)?

Reply:
Maybe something that takes the url as argument, returns the result and use exceptions for errors.

A simple implementation would be
```
function (string $url): string {
    $result = @file_get_contents($url);

    if (false === $result) {
        throw new RuntimeException();
    }

    return $result
}
```

Reply:
@jvasseur 👍, I'll update the comment to remove `false` from the signature.